src/main/scala/viper/gobra/frontend/info/implementation/property/ConstantEvaluation.scala

```diff
@@ -64,7 +66,27 @@ trait ConstantEvaluation { this: TypeInfoImpl =>
 
   lazy val intConstantEval: PExpression => Option[BigInt] =
     attr[PExpression, Option[BigInt]] {
-      case PIntLit(lit) => Some(lit)
+      case PIntLit(lit, _) => Some(lit)
+      case inv: PInvoke => resolve(inv) match {
+        case Some(conv: ap.Conversion) => underlyingTypeP(conv.typ) match {
+          case Some(_: PIntegerType) => intConstantEval(conv.arg)
+          case _ => None
+        }
+        case _ => None
+      }
+      case PBitNegation(op) =>
+        // Not sufficient to do `intConstantEval(op) map (_.unary_~)`, produces wrong results for unsigned int values
+        exprType(op) match {
+          case IntT(t) =>
+            val constEval = intConstantEval(op)
+            constEval map { constValue =>
+              t match {
+                case UnboundedInteger | _: Signed => ~constValue
+                case u: Unsigned => BigInt(~constValue.toShort) mod (u.upper + 1)
```

Comment:
why do you call `toShort`? Doesn't this result in a loss of precision?

Reply:
surprisingly, I did not manage to trigger a case where this caused loss of precision. I have however changed the code to be less fragile in this regard

Reply:
But why does it has to be a short? Elsewhere in this file, you're for example using `toLong`

Reply:
Long story short, it does not need to be cast actually, `mod` guarantees the correct result without it. However, in my initial version (before using `mod`), I did need a cast to be sure that I produced the correct result. More precisely, I wanted to invert the exact number of bits that the bitwise representation has. I forgot to remove it when I started using `mod`. I removed this complexity in my last commit.